task/control-plane-agent/src/mgs_gimlet.rs

```diff
@@ -908,6 +917,46 @@ impl SpHandler for MgsHandler {
             .map_err(|_| SpError::Busy)?;
         Ok(())
     }
+
+    fn set_ipcc_key_lookup_value(
+        &mut self,
+        _sender: SocketAddrV6,
+        _port: SpPort,
+        key: u8,
+        value: &[u8],
+    ) -> Result<(), SpError> {
+        use gateway_messages::IpccKeyLookupValueError;
+        use host_sp_messages::Key;
+
+        ringbuf_entry!(Log::MgsMessage(MgsMessage::SetIpccKeyValue {
+            key,
+            value_len: value.len(),
+        }));
+
+        match Key::from_u8(key) {
+            Some(Key::InstallinatorImageId) => {
+                if value.len() > self.installinator_image_id.capacity() {
+                    return Err(SpError::SetIpccKeyLookupValueFailed(
+                        IpccKeyLookupValueError::ValueTooLong {
+                            max_len: self.installinator_image_id.capacity()
+                                as u16,
+                        },
+                    ));
+                }
```

Comment:
Nit: maybe extract `let max_len = self.installinator_image_id.capacity()` before this conditional, which helps with the wacky indentation below?

```suggestion
                let max_len = installinator_image_id.capacity();
                if value.len() > max_len {
                    return Err(SpError::SetIpccKeyLookupValueFailed(
                        IpccKeyLookupValueError::ValueTooLong {
                            max_len: max_len as u16,
                        },
                    ));
                }
```